android/app/src/main/java/app/organicmaps/car/screens/bookmarks/BookmarksLoader.java

```diff
@@ -0,0 +1,124 @@
+package app.organicmaps.car.screens.bookmarks;
+
+import android.graphics.drawable.Drawable;
+import android.location.Location;
+import android.text.SpannableStringBuilder;
+import android.text.Spanned;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.car.app.CarContext;
+import androidx.car.app.constraints.ConstraintManager;
+import androidx.car.app.model.CarIcon;
+import androidx.car.app.model.DistanceSpan;
+import androidx.car.app.model.ForegroundCarColorSpan;
+import androidx.car.app.model.ItemList;
+import androidx.car.app.model.Row;
+import androidx.core.graphics.drawable.IconCompat;
+
+import app.organicmaps.R;
+import app.organicmaps.bookmarks.data.BookmarkCategory;
+import app.organicmaps.bookmarks.data.BookmarkInfo;
+import app.organicmaps.bookmarks.data.BookmarkManager;
+import app.organicmaps.bookmarks.data.Icon;
+import app.organicmaps.car.util.Colors;
+import app.organicmaps.car.util.RoutingHelpers;
+import app.organicmaps.location.LocationHelper;
+import app.organicmaps.util.Distance;
+import app.organicmaps.util.Graphics;
+import app.organicmaps.util.concurrency.ThreadPool;
+import app.organicmaps.util.concurrency.UiThread;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class BookmarksLoader
+{
+  public interface OnBookmarksLoaded
+  {
+    void onBookmarksLoaded(@NonNull ItemList bookmarks);
+  }
+
+  // The maximum size should be equal to ConstraintManager.CONTENT_LIMIT_TYPE_LIST.
+  // However, having more than 50 items results in android.os.TransactionTooLargeException.
+  // This exception occurs because the data parcel size is too large to be transferred between services.
+  // The primary cause of this issue is the icons. Even though we have the maximum Icon.TYPE_ICONS.length icons,
+  // each row contains a unique icon, resulting in serialization of each icon.
```

Comment:
What are the ways to avoid passing equal icons?

Reply:
No way. It's not our code.
Yout can try to create a ticket on google's issue tracker and wait for 5 years until they fix this 😄